egui/src/grid.rs

```diff
@@ -336,16 +491,46 @@ impl Grid {
         // which we do here:
         ui.horizontal(|ui| {
             let id = ui.make_persistent_id(id_source);
-            let grid = GridLayout {
-                striped,
+            let mut grid = GridLayout {
                 spacing,
                 min_cell_size: vec2(min_col_width, min_row_height),
                 max_cell_size,
+                color_spec,
                 row: start_row,
                 ..GridLayout::new(ui, id)
             };
 
+            // Set convenience color specs
+            if striped {
+                grid.color_spec.row_pickers.push(|row| {
+                    if row % 2 == 0 {
+                        Some(GuiColor::light_dark(
+                            Rgba::from_black_alpha(0.075),
+                            Rgba::from_white_alpha(0.0075),
+                        ))
+                    } else {
+                        None
+                    }
+                });
+            }
+
+            if header_row {
+                grid.color_spec.row_pickers.push(|row| {
+                    if row == 0 {
+                        Some(GuiColor::light_dark(
+                            Rgba::from_black_alpha(0.75),
+                            Rgba::from_white_alpha(0.075),
+                        ))
+                    } else {
+                        None
+                    }
+                });
+            }
+
+            // ---
+
             ui.set_grid(grid);
+            ui.start_row();
```

Comment:
Instead of adding two `fn start_row`, just have `Placer::set_grid` call the correct function directly. In this case I *believe* what you are doing is painting the top row?

Reply:
I don't think `set_grid` has access to the Painter, because that's passed in when calling grid-related functions from `Ui`. I may be mistaken.